src/codegeneration/ES6ClassTransformer.js

```diff
@@ -0,0 +1,270 @@
+    } else {
+      if (this.options_.memberVariables) {
+        constructor = this.appendInstanceInitializers_(constructor,
+            initStatements, tree.superClass);
+      }
+    }
+
+    elements.push(constructor);
+
+    return {
+      elements,
+      initStaticVars,
+    };
+  }
+
+  /**
+   * Transforms a single class declaration
+   *
+   * @param {ClassDeclaration} tree
+   * @return {ParseTree}
+   */
+  transformClassDeclaration(tree) {
+    let {
+      elements,
+      initStaticVars,
+    } = this.transformClassElements_(tree);
+
+    let statements = createStaticInitializerStatements(tree.name, initStaticVars);
+
+    statements = prependStatements(statements,
+        createClassDeclaration(tree.name, tree.superClass, elements));
+
+    return new AnonBlock(null, statements);
+  }
+
+  /**
+   * Transforms a single class expression
+   *
+   * @param {ClassExpression} tree
+   * @return {ParseTree}
+   */
+  transformClassExpression(tree) {
+    let {
+      elements,
+      initStaticVars,
+    } = this.transformClassElements_(tree);
+
+    let classExpression = new ClassExpression(tree.location, tree.name,
+        tree.superClass, elements, tree.annotations, tree.typeParameters);
+
+    if (initStaticVars.length === 0) {
+      return classExpression;
+    }
+
+    let id = this.idGenerator_.generateUniqueIdentifier();
+    let className = createBindingIdentifier(id);
+    let statements = [];
+
+    statements.push(parseStatement `let ${id} = ${classExpression}`);
+    statements = statements.concat(createStaticInitializerStatements(className, initStaticVars));
+    statements.push(parseStatement `return ${className}`);
+
+    let body = createFunctionBody(statements);
+
+    return createImmediatelyInvokedFunctionExpression(body);
+  }
+
+
+  getDefaultConstructor_(tree, initStatements) {
+    let constructorParams = createEmptyParameterList();
+    let constructorBody;
+    if (tree.superClass) {
+      let statement = parseStatement `super(...arguments)`;
+      constructorBody = createFunctionBody([statement, ...initStatements]);
+    } else {
+      constructorBody = createFunctionBody(initStatements);
+    }
+
+    let name = new LiteralPropertyName(null, createIdentifierToken(CONSTRUCTOR));
+
+    return new PropertyMethodAssignment(tree.location, false,
+        null, name, constructorParams, null, [], constructorBody);
+  }
+
+  /**
+   *  Appends assignment to this for the member variables after the super call.
+   */
+  appendInstanceInitializers_(constructor, initStatements, superClass) {
+    if (initStatements.length === 0) return constructor;
+
+    let statements = constructor.body.statements;
+
+    if (superClass) {
+      let superExpressionIndex = -1;
+
+      for (let i = 0; i < statements.length; i++) {
+        let statement = statements[i];
+        if (statement.isDirectivePrologue()) {
+          continue;
+        }
+        if (isSuper(statement)) {
```

Comment:
This should be addressed in the latest commit where I've added a proper transformer.

I was also able to do some refactoring / cleaning because we now know that ` super()`  is always here for derived class